compiler/z/codegen/OMRInstruction.cpp

```diff
@@ -683,37 +700,50 @@ static void handleLoadWithRegRanges(TR::Instruction *inst, TR::CodeGenerator *cg
          opCodeToUse == TR::InstOpCode::LMY || opCodeToUse == TR::InstOpCode::VLM))
       return;
 
-   bool isVector = opCodeToUse == TR::InstOpCode::VLM ? true : false;
-   uint32_t numVRFs = TR::RealRegister::LastAssignableVRF - TR::RealRegister::FirstAssignableVRF;
+   TR::Register *lowReg, *highReg;
+   uint32_t minRegNum, maxRegNum;
+   uint32_t availForShuffle;
 
-   TR::Register *lowReg  = isVector ? toS390VRSInstruction(inst)->getFirstRegister()  : toS390RSInstruction(inst)->getFirstRegister();
-   TR::Register *highReg = isVector ? toS390VRSInstruction(inst)->getSecondRegister() : toS390RSInstruction(inst)->getSecondRegister();
+   if (opCodeToUse == TR::InstOpCode::VLM)
+      {
+      lowReg = toS390VRSInstruction(inst)->getFirstRegister();
+      highReg = toS390VRSInstruction(inst)->getSecondRegister();
```

Comment:
When using Vector Load Multiple, we should never have `highReg` lower than `lowReg`. Wrapping around is only available for GPRs (For example if lowReg is GPR2 and highReg is GPR1, it will load all registers starting with GPR2 and second last and last element will be loaded into GPR0 and GPR1 respectively). 
We should have an ASSERT in case we ever try to use VLM similarly.

Another thing I observed looking into reference of VLM instruction, is that, maximum number of registers they can handle are at most 16. If it is higher, then specific exception is recognized. We should check for that as well.